rust/kcl-lib/src/execution/exec_ast.rs

```diff
@@ -2401,6 +2402,24 @@ p = {
                 .unwrap(),
             KclValue::Plane { .. }
         ));
+        let arr1 = mem
+            .memory
+            .get_from("arr1", result.mem_env, SourceRange::default(), 0)
+            .unwrap();
+        if let KclValue::HomArray { value, ty } = arr1 {
+            assert_eq!(value.len(), 1, "Expected Vec with specific length: found {:?}", value);
+            assert_eq!(*ty, RuntimeType::known_length(UnitLen::Cm));
+            // Compare, ignoring meta.
+            if let KclValue::Number { value, ty, .. } = &value[0] {
+                // Converted from mm to cm.
+                assert_eq!(*value, 4.2);
```

Comment:
Hmm, this seems bad, but its not your fault. For numeric types, type ascription does not convert the units, but it does for arrays of numeric types.

Reply:
I have an idea for how to fix this, but I'd like to get this PR merged first.